device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsAuthenticationLinkHandler.java

```diff
@@ -0,0 +1,71 @@
+package com.microsoft.azure.sdk.iot.device.transport.amqps;
+
+import com.microsoft.azure.sdk.iot.device.DeviceClientConfig;
+import com.microsoft.azure.sdk.iot.device.Message;
+import com.microsoft.azure.sdk.iot.device.exceptions.TransportException;
+import com.microsoft.azure.sdk.iot.device.transport.IotHubTransportMessage;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.qpid.proton.Proton;
+import org.apache.qpid.proton.ProtonUnsupportedOperationException;
+import org.apache.qpid.proton.engine.SslDomain;
+import org.apache.qpid.proton.engine.Transport;
+import org.apache.qpid.proton.message.impl.MessageImpl;
+
+import javax.net.ssl.SSLContext;
+import java.util.UUID;
+
+@Slf4j
+public abstract class AmqpsAuthenticationLinkHandler extends AmqpsLinksHandler
```

Comment:
Why the trailing 's' in Amqps?

Reply:
AMQPS is to AMQP what HTTPS is to HTTP. It just signifies that we are doing ssl handshakes when connecting